File: frontend/src/interfaces/SegmentModel.ts
import KeyValueModel from "./KeyValueModel";
import SegmentPreviewModel from "./SegmentPreviewModel";

interface SegmentModel extends SegmentPreviewModel {
  heartRateGraph: KeyValueModel<number, number | null>[];
  paceGraph: KeyValueModel<number, number>[];
  altitudeGraph: KeyValueModel<number, number | null>[];
  cadenceGraph: KeyValueModel<number, number | null>[];
}

export default SegmentModel;

File: backend/src/Endurist.Common/Models/Activities/SegmentModel.cs
using SideEffect.DataTransfer.Models;

namespace Endurist.Common.Models.Activities;

public class SegmentModel : SegmentPreviewModel
{
    public List<KeyValueModel<double, int?>> HeartRateGraph { get; set; }

    public List<KeyValueModel<double, int>> PaceGraph { get; set; }

    public List<KeyValueModel<double, double?>> AltitudeGraph { get; set; }

    public List<KeyValueModel<double, int?>> CadenceGraph { get; set; }
}
